rpc/legacyrpc/methods.go

```diff
@@ -208,64 +179,40 @@ type lazyHandler func() (interface{}, *dcrjson.RPCError)
 // returning a closure that will execute it with the (required) wallet and
 // (optional) consensus RPC server.  If no handlers are found and the
 // chainClient is not nil, the returned handler performs RPC passthrough.
-func lazyApplyHandler(request *dcrjson.Request, w *wallet.Wallet,
-	chainClient *dcrrpcclient.Client, l *loader.Loader) lazyHandler {
-
-	handlerData, ok := rpcHandlers[request.Method]
-	if ok && handlerData.handlerWithChain != nil && w != nil && chainClient != nil {
-		return func() (interface{}, *dcrjson.RPCError) {
-			cmd, err := dcrjson.UnmarshalCmd(request)
-			if err != nil {
-				return nil, dcrjson.ErrRPCInvalidRequest
-			}
-			resp, err := handlerData.handlerWithChain(cmd, w, chainClient)
-			if err != nil {
-				return nil, jsonError(err)
-			}
-			return resp, nil
-		}
-	}
-	if ok && handlerData.handlerWithLoader != nil && w != nil {
+func lazyApplyHandler(s *Server, request *dcrjson.Request) lazyHandler {
+	handlerData, ok := handlers[request.Method]
+	if !ok {
 		return func() (interface{}, *dcrjson.RPCError) {
-			cmd, err := dcrjson.UnmarshalCmd(request)
-			if err != nil {
-				return nil, dcrjson.ErrRPCInvalidRequest
+			if s.chainClient == nil {
+				return nil, &dcrjson.RPCError{
+					Code:    -1,
+					Message: "Chain RPC is inactive",
+				}
 			}
-			resp, err := handlerData.handlerWithLoader(cmd, w, l)
+			resp, err := s.chainClient.RawRequest(request.Method, request.Params)
 			if err != nil {
 				return nil, jsonError(err)
 			}
-			return resp, nil
+			return &resp, nil
 		}
 	}
-	if ok && handlerData.handler != nil && w != nil {
+
+	if s.chainClient != nil {
 		return func() (interface{}, *dcrjson.RPCError) {
 			cmd, err := dcrjson.UnmarshalCmd(request)
 			if err != nil {
 				return nil, dcrjson.ErrRPCInvalidRequest
 			}
-			resp, err := handlerData.handler(cmd, w)
+
+			resp, err := handlerData.fn(s, cmd)
 			if err != nil {
 				return nil, jsonError(err)
 			}
 			return resp, nil
 		}
 	}
 
-	// Fallback to RPC passthrough
-	return func() (interface{}, *dcrjson.RPCError) {
-		if chainClient == nil {
-			return nil, &dcrjson.RPCError{
-				Code:    -1,
-				Message: "Chain RPC is inactive",
-			}
-		}
-		resp, err := chainClient.RawRequest(request.Method, request.Params)
-		if err != nil {
-			return nil, jsonError(err)
-		}
-		return &resp, nil
-	}
+	return nil
```

Comment:
these are some RPCs that can be called that don't require the chain client at all (all the previous ones that had the `handler` specified for them but not the `handlerWithChain` function).  Those wouldn't be callable anymore here.  It's also an unsafe nil usage since the function being returned here must always do something, either perform the RPC or return an error.